orchestrator/ethereum_gravity/src/send_erc721_to_cosmos.rs

```diff
@@ -0,0 +1,115 @@
+//! Helper functions for sending tokens to Cosmos
+
+use clarity::abi::{encode_call, Token};
+use clarity::PrivateKey as EthPrivateKey;
+use clarity::{Address, Uint256};
+use deep_space::address::Address as CosmosAddress;
+use gravity_utils::error::GravityError;
+use std::time::{Duration, Instant};
+use web30::client::Web3;
+use web30::types::SendTxOption;
+
+pub const SEND_TO_COSMOS_GAS_LIMIT: u128 = 100_000;
+ 
+#[allow(clippy::too_many_arguments)]
+pub async fn send_erc721_to_cosmos(
+    erc20: Address,
+    gravity_contract: Address,
+    amount: Uint256,
+    cosmos_destination: CosmosAddress,
+    sender_secret: EthPrivateKey,
+    wait_timeout: Option<Duration>,
+    web3: &Web3,
+    options: Vec<SendTxOption>,
+) -> Result<Uint256, GravityError> {
+    print!("Now in the send_erc721_to_cosmos function!");
+    let sender_address = sender_secret.to_address();
+    let mut approve_nonce = None;
+
+    for option in options.iter() {
+        if let SendTxOption::Nonce(_) = option {
+            return Err(GravityError::InvalidOptionsError(
+                "This call sends more than one tx! Can't specify".to_string(),
+            ));
+        }
+    }
+
+    // rapidly changing gas prices can cause this to fail, a quick retry loop here
+    // retries in a way that assists our transaction stress test
+    let mut approved = web3
+        .check_erc20_approved(erc20, sender_address, gravity_contract)
+        .await;
+    if let Some(w) = wait_timeout {
+        let start = Instant::now();
+        // keep trying while there's still time
+        while approved.is_err() && Instant::now() - start < w {
+            approved = web3
+                .check_erc20_approved(erc20, sender_address, gravity_contract)
```

Comment:
this may or may not work. I think the function header is the same so it might. There should be a check erc721 approved function in web30 for this.